packages/fast-tooling-react/src/css-editor/box-shadow/box-shadow.tsx

```diff
@@ -0,0 +1,209 @@
+                        />
+                    </div>
+                </div>
+                <div
+                    className={get(
+                        this.props,
+                        "managedClasses.cssBoxShadow_controlRegion"
+                    )}
+                >
+                    <label
+                        className={get(this.props, "managedClasses.cssBoxShadow_label")}
+                    >
+                        X
+                    </label>
+                    <input
+                        className={get(this.props, "managedClasses.cssBoxShadow_xInput")}
+                        type={"text"}
+                        placeholder={"5px"}
+                        value={this.state.boxShadowOffsetX}
+                        onChange={this.handleBoxShadowXOnChange}
+                    />
+                </div>
+                <div
+                    className={get(
+                        this.props,
+                        "managedClasses.cssBoxShadow_controlRegion"
+                    )}
+                >
+                    <label
+                        className={get(this.props, "managedClasses.cssBoxShadow_label")}
+                    >
+                        Y
+                    </label>
+                    <input
+                        className={get(this.props, "managedClasses.cssBoxShadow_yInput")}
+                        type={"text"}
+                        placeholder={"5px"}
+                        value={this.state.boxShadowOffsetY}
+                        onChange={this.handleBoxShadowYOnChange}
+                    />
+                </div>
+                <div
+                    className={get(
+                        this.props,
+                        "managedClasses.cssBoxShadow_controlRegion"
+                    )}
+                >
+                    <label
+                        className={get(this.props, "managedClasses.cssBoxShadow_label")}
+                    >
+                        Blur
+                    </label>
+                    <input
+                        className={get(
+                            this.props,
+                            "managedClasses.cssBoxShadow_blurInput"
+                        )}
+                        type={"text"}
+                        placeholder={"0"}
+                        value={this.state.boxShadowBlurRadius}
+                        onChange={this.handleBoxShadowBlurOnChange}
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    private handleBoxShadowColorOnChange = (
+        e: React.ChangeEvent<HTMLInputElement>
+    ): void => {
+        this.setState({
+            boxShadowColor: e.target.value,
+        });
+    };
+
+    private handleBoxShadowOpacityOnChange = (
+        e: React.ChangeEvent<HTMLInputElement>
+    ): void => {
+        this.setState({
+            boxShadowOpacity: e.target.valueAsNumber,
```

Comment:
these all need to store default values and use them as a fallback, otherwise this can be undefined and crash